microsoft-azure-storage/src/com/microsoft/azure/storage/core/BaseRequest.java

```diff
@@ -64,6 +65,11 @@
      */
     private static String userAgent;
 
+    /**
+     * Singleton instance for keepalive Socket Factory
+     */
+    private static SSLSocketFactory staticKeepAliveSocketFactory ;
```

Comment:
Instead of putting it here, I think that we can put it in the KeepAliveSocketFactory and have getInstance() method. This is effectively following the singleton pattern.

Reply:
Good point, thanks, addressed.